server/ERNI.PhotoDatabase.Server/Controllers/PhotoController.cs

```diff
@@ -8,77 +8,88 @@
 using ERNI.PhotoDatabase.DataAccess.Images;
 using ERNI.PhotoDatabase.DataAccess.Repository;
 using ERNI.PhotoDatabase.DataAccess.UnitOfWork;
+using ERNI.PhotoDatabase.Server.Configuration;
 using ERNI.PhotoDatabase.Server.Utils.Image;
 using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
 
 namespace ERNI.PhotoDatabase.Server.Controllers
 {
 
     [Route("api/[controller]")]
     public class PhotoController : Controller
     {
-        private readonly IPhotoRepository repository;
-        private readonly ImageStore imageStore;
-        private readonly IUnitOfWork unitOfwork;
+        private readonly Lazy<IPhotoRepository> repository;
+        private readonly Lazy<ImageStore> imageStore;
+        private readonly Lazy<IUnitOfWork> unitOfwork;
+        private readonly Lazy<IImageManipulation> imageTools;
+        private readonly IOptions<ImageSizesSettings> settings;
 
```

Comment:
I am fine with Lazy, but when it comes to usage, I don't like `repository.Value.Somthing` pattern. Maybe we can hide Lazy fields behind property. ... just idea.

Reply:
Wouldn't it be awesome if it could be used like a monad? :D Now I have to create properties for those.